command.go

```diff
@@ -74,6 +74,12 @@ func (c *Command) predict(a Args) (options []string, only bool) {
 				return
 			}
 		}
+
+		// We matched so stop searching. Continuing to search can accidentally
+		// match a subcommand with current set of commands, see issue #46.
+		if subCommandFound {
```

Comment:
Nice catch! good comment!
Can you just remove the if and put it in the if statement above, after line 75.
In the beginning of the if statement we set the `subCommandFound` so you can break at the end of the `if` statement.